spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/filter/headers/observation/DefaultGatewayObservationConvention.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2013-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gateway.filter.headers.observation;
+
+import io.micrometer.common.KeyValues;
+import io.micrometer.common.lang.NonNull;
+import io.micrometer.common.lang.Nullable;
+
+import static org.springframework.cloud.gateway.filter.headers.observation.GatewayDocumentedObservation.LowCardinalityKeys.METHOD;
+import static org.springframework.cloud.gateway.filter.headers.observation.GatewayDocumentedObservation.LowCardinalityKeys.STATUS;
+import static org.springframework.cloud.gateway.filter.headers.observation.GatewayDocumentedObservation.LowCardinalityKeys.URI;
+
+public class DefaultGatewayObservationConvention implements GatewayObservationConvention {
+
+	public static final GatewayObservationConvention INSTANCE = new DefaultGatewayObservationConvention();
+
+	@Override
+	public KeyValues getLowCardinalityKeyValues(GatewayContext context) {
+		KeyValues keyValues = KeyValues.empty();
+		if (context.getCarrier() == null) {
+			return keyValues;
+		}
+		// TODO: URI is high cardinality
+		keyValues = keyValues.and(METHOD.withValue(context.getRequest().getMethod().name()), URI.withValue(context.getRequest().getURI().toString()));
+		if (context.getResponse() != null && context.getResponse().getStatusCode() != null) {
+			keyValues = keyValues.and(STATUS.withValue(String.valueOf(context.getResponse().getStatusCode().value())));
```

Comment:
This can lead to an interesting issue, please bear with me.
If the status code does not present (e.g.: timeout?) this will not add a status key-value which means that in those scenarios this tag will be missing from the `Timer`.

This can cause issues in certain backends (i.e.: Prometheus) since it will lead to time series where the name is the same but the tags are different, e.g.:

```
http_client_requests{method="GET", status="200"} ...
http_client_requests{method="GET"} ...
```

So I guess my question is: will the status code always be there if we already initiated the call (so we are not handling the LongTaskTimer case)?
If so, with some default value (`0`, `-1`), we are good. But if not, I would do this:
```
else {
    keyValues = keyValues.and(STATUS.withValue("n/a"));
}
```